src/main/java/edu/ucsb/cs56/mapache_search/UserController.java

```diff
@@ -35,23 +40,34 @@ public String index(Model model) {
         model.addAttribute("users", users);
         return "user/index";
     }
-
+    
     @GetMapping("user/settings")
     public String settingsForm(Model model, OAuth2AuthenticationToken token) {
-
         AppUser u = userRepository.findByUid(controllerAdvice.getUid(token)).get(0);
+        Long searches = userRepository.findByUid(controllerAdvice.getUid(token)).get(0).getSearches();
+        Long maxsearches = userRepository.findByUid(controllerAdvice.getUid(token)).get(0).getMaxsearches();
+        Long time = userRepository.findByUid(controllerAdvice.getUid(token)).get(0).getTime();
         model.addAttribute("user", u);
         model.addAttribute("user_template", new AppUser());
+        model.addAttribute("api_uses", searches);
+        model.addAttribute("time", time);
+        model.addAttribute("max_uses", maxsearches);
         return "user/settings";
     }
 
     @PostMapping("user/settings/update")
     public String updateKey(@ModelAttribute AppUser user, Model model, OAuth2AuthenticationToken token) {
         AppUser u = userRepository.findByUid(controllerAdvice.getUid(token)).get(0);
+        Long searches = userRepository.findByUid(controllerAdvice.getUid(token)).get(0).getSearches();
+        Long maxsearches = userRepository.findByUid(controllerAdvice.getUid(token)).get(0).getMaxsearches();
```

Comment:
Since maxsearches is a constant, this can be done more simply by making it public.  Then you can access `AppUser.maxsearches` directly as a value.